backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/predicate/impl/AbstractLuceneSearchPredicateBuilder.java

```diff
@@ -38,15 +39,16 @@ public LuceneSearchPredicateBuilder toImplementation() {
 
 	@Override
 	public final Query build(LuceneSearchPredicateContext context) {
-
-		// TODO handle withConstantScore value here!
+		Query query = doBuild( context );
 
 		if ( boost != null ) {
-			return new BoostQuery( doBuild( context ), boost );
+			query = new BoostQuery( query, boost );
 		}
-		else {
-			return doBuild( context );
+		if ( withConstantScore ) {
+			query = new ConstantScoreQuery( query );
```

Comment:
I just saw this, I didn't notice during my first review, sorry: the boost should be applied on top of the constant score, otherwise the boost will simply be ignored.

I see the tests passed, so maybe you're missing a test that checks the boost is correctly applied in conjunction with ConstantScore? I'd recommend to work on the test before the fix, so you can check the test is effective.

A simple way to test this would be to execute a query with two predicates, each matching a different document, each with a constant score and a different boost. Swap the boosts, the result order should be reversed.

Reply:
A bug is also a chance to add another good test. In fact, you're right. I added the test you described and I got an error from Lucene backend. Thanks